File: regexlib/2021-5-15/javascript_test_file/regexlib_7847.js
/* 7847
 * [0-9]*\.?[0-9]*[1-9]
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"0"*1024+"◎@! _1SLQ_1"
 */
var REGEX = /[0-9]*\.?[0-9]*[1-9]/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '0'.repeat(i*1) + '◎@! _1SLQ_1'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_7847.py
# 7847
# [0-9]*\.?[0-9]*[1-9]
# EXPONENT
# nums:4
# EXPONENT AttackString:""+"0"*1024+"◎@! _1SLQ_1"

import re
from time import perf_counter

regex = """[0-9]*\.?[0-9]*[1-9]"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "0" * i * 1 + "◎@! _1SLQ_1"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")